dcm4che-net-audit/src/test/java/org/dcm4che3/net/audit/AuditLoggerTest.java

```diff
@@ -0,0 +1,356 @@
+
+    @Test
+    public void reconfigure_UpdatesSpoolFileNameSuffix_WhenCalled() {
+        newLogger.setSpoolFileNameSuffix("POST");
+        logger.setSpoolFileNameSuffix("PRE");
+
+        logger.reconfigure(newLogger);
+
+        assertThat("spoolFileNameSuffix is not as expected", logger.getSpoolFileNameSuffix(), is("POST"));
+    }
+
+    @Test
+    public void reconfigure_UpdatesRetryInterval_WhenCalled() {
+        newLogger.setRetryInterval(10);
+        logger.setRetryInterval(1);
+
+        logger.reconfigure(newLogger);
+
+        assertThat("retryInterval is not as expected", logger.getRetryInterval(), is(10));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditLoggerInstalled_WhenCalled() {
+        newLogger.setAuditLoggerInstalled(false);
+        logger.setAuditLoggerInstalled(true);
+
+        logger.reconfigure(newLogger);
+
+        assertThat("auditLoggerInstalled is not as expected", logger.getAuditLoggerInstalled(), is(false));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditRecordRepositoryDevices_WhenCalled() {
+        final Connection auditUDP = new Connection("audit-udp", "localhost", 514);
+        auditUDP.setProtocol(Connection.Protocol.SYSLOG_UDP);
+
+        final AuditRecordRepository arr1 = new AuditRecordRepository();
+        arr1.addConnection(auditUDP);
+
+        final Device arrDevice1 = new Device("syslog");
+        arrDevice1.addDeviceExtension(arr1);
+        arrDevice1.addConnection(auditUDP);
+
+        newLogger.setAuditRecordRepositoryDevices(Collections.singletonList(arrDevice1));
+
+        final Connection auditTCP = new Connection("audit-tcp", "localhost", 516);
+        auditTCP.setProtocol(Connection.Protocol.SYSLOG_TLS);
+
+        final AuditRecordRepository arr2 = new AuditRecordRepository();
+        arr2.addConnection(auditTCP);
+
+        final Device arrDevice2 = new Device("syslog2");
+        arrDevice2.addDeviceExtension(arr2);
+        arrDevice2.addConnection(auditTCP);
+
+        logger.setAuditRecordRepositoryDevices(Collections.singletonList(arrDevice2));
+
+        logger.reconfigure(newLogger);
+
+        final Device toDevice = logger.getAuditRecordRepositoryDevices().get(0);
+        assertThat("auditRecordRepositoryDevices' device name is not as expected", toDevice.getDeviceName(), is(arrDevice1.getDeviceName()));
+        assertThat("auditRecordRepositoryDevices' device extension is not as expected", toDevice.getDeviceExtension(AuditRecordRepository.class), samePropertyValuesAs(arr1));
+        assertThat("auditRecordRepositoryDevices' connections is not as expected", toDevice.getConnections().get(0), samePropertyValuesAs(auditUDP));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditSuppressCriteriaList_WhenCalled() {
+        final AuditSuppressCriteria newCriteria = new AuditSuppressCriteria();
+        newCriteria.setCommonName("NewCommonName");
+        newLogger.setAuditSuppressCriteriaList(Collections.singletonList(newCriteria));

```

Comment:
Is there a benefit to using singletonList vs JDK Arrays.asList()?

Reply:
If you only have one item in the list, SonarLint prefers using singletonList, not Arrays.asList().